cli/src/commands/fix.rs

```diff
@@ -0,0 +1,320 @@
+// Copyright 2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{HashMap, HashSet};
+use std::io::Write;
+use std::process::{Command, Stdio};
+use std::sync::mpsc::channel;
+use std::sync::Arc;
+
+use futures::StreamExt;
+use itertools::Itertools;
+use jj_lib::backend::{BackendError, BackendResult, CommitId, FileId, TreeValue};
+use jj_lib::commit::{Commit, CommitIteratorExt};
+use jj_lib::matchers::EverythingMatcher;
+use jj_lib::merged_tree::MergedTreeBuilder;
+use jj_lib::repo::Repo;
+use jj_lib::repo_path::RepoPathBuf;
+use jj_lib::revset::{RevsetExpression, RevsetIteratorExt};
+use jj_lib::store::Store;
+use pollster::FutureExt;
+use rayon::iter::IntoParallelIterator;
+use rayon::prelude::ParallelIterator;
+use tracing::instrument;
+
+use crate::cli_util::{CommandHelper, RevisionArg, WorkspaceCommandTransaction};
+use crate::command_error::CommandError;
+use crate::ui::Ui;
+
+/// Update files with formatting fixes or other changes
+///
+/// The primary use case for this command is to apply the results of automatic
+/// code formatting tools to revisions that may not be properly formatted yet.
+/// It can also be used to modify files with other tools like `sed` or `sort`.
+///
+/// The changed files in the given revisions will be updated with any fixes
+/// determined by passing their file content through the external tool.
+/// Descendants will also be updated by passing their versions of the same files
+/// through the same external tool, which will never result in new conflicts.
+/// Files with existing conflicts will be updated on all sides of the conflict,
+/// which can potentially increase or decrease the number of conflict markers.
+///
+/// The external tool must accept the current file content on standard input,
+/// and return the updated file content on standard output. The output will not
+/// be used unless the tool exits with a successful exit code. Output on
+/// standard error will be ignored.
+#[derive(clap::Args, Clone, Debug)]
+pub(crate) struct FixArgs {
+    /// Fix files in the specified revision(s) and their descendants
+    #[arg(long, short)]
+    sources: Vec<RevisionArg>,
```

Comment:
Shouldn't it be `-r`/`--revision` for consistency?

Reply:
It's `-s/--sources` because it includes descendants, just like `jj rebase -s`.

btw, the long form of `jj rebase -s` is singular `--source`, so we should probably make that consistent.